apps/bare-expo/android/app/build.gradle

```diff
@@ -219,3 +219,45 @@ task copyDownloadableDepsToLibs(type: Copy) {
     from configurations.compile
     into 'libs'
 }
+
+/**
+ * Download a precompiled version of React Native
+ */
+buildscript {
+  repositories {
+    jcenter()
+  }
+  dependencies {
+    classpath 'de.undercouch:gradle-download-task:4.0.0'
+  }
+}
+apply plugin: "de.undercouch.download"
+def downloadsDir = new File(buildDir, "downloads")
+
+afterEvaluate {
+  downloadsDir.mkdirs()
+  // Fetch from NPM
+  def packageUrl = "https://registry.npmjs.org/react-native/-/react-native-${reactNativeVersion}.tgz"
+  def packageDest = new File(downloadsDir, "react-native_${reactNativeVersion}.tar.gz")
+  download {
+    src(packageUrl)
+    overwrite false
+    tempAndMove true
+    dest(packageDest)
+  }
+  copy {
+    from(tarTree(resources.gzip(packageDest.toString()))) {
+      include "package/android/**"
+      eachFile {
+        fcd -> fcd.relativePath = new RelativePath(!fcd.isDirectory(), fcd.relativePath.segments.drop(2))
+      }
+    }
+    into(downloadsMavenRepo)
+  }
+}
+
+task cleanDownloadedReactNative {
+  new File(downloadsMavenRepo).deleteDir()
+}
+
+clean.dependsOn(cleanDownloadedReactNative)
```

Comment:
Isn't there any other less complicated solution than writing custom `gradle script` for downloading original `react-native` package/module? I mean, there are/or could be others projects under `apps/` than need original `react-native`.
Cannot we solve it somehow on `yarn/lerna` level (but from what I know and see there's no such solution 😢)?

Reply:
In an ideal world there is only one project requiring original `react-native` package (why would we want to maintain multiple projects?).

I know this isn't the most elegant solution one could think of and it has downsides (JS code comes from `-lab` as does the iOS code; on the other hand, changes in those two don't break entire apps), but I haven't found any better solution available and this is simple enough that we have a working solution in a couple of lines. 🙂

We could also write a JS script that would download and extract the AAR when installing the JS dependencies, but… why move it to JS side if it belongs to Gradle lifecycle?